components/web_discovery/browser/double_fetcher.cc

```diff
@@ -88,24 +94,56 @@ void DoubleFetcher::OnFetchTimer(const base::Value& request_data) {
   }
 
   GURL url(*url_str);
+
+  if (base::Contains(url.host_piece(), kGoogleSLD) &&
```

Comment:
wouldn't it match `https://schmoogoogle.com` ? is it deliberate?

Reply:
keep in mind that we have an allowlist of URLs to scrape defined in the WDP manifest/patterns, as well as an allowlist in the `BackupResultsService`, so this is very unlikely. I decided to use `base::Contains` to keep this simple since we already have these more robust mechanisms in place

Reply:
it just looks strange. May i ask @fmarier to say this code is fine to merge

Reply:
amended to use the allowed url function